tst/EnergyPlus/unit/MixedAir.unit.cc

```diff
@@ -874,14 +912,359 @@ TEST_F(EnergyPlusFixture, CO2ControlDesignOccupancyTest)
     EXPECT_EQ("ProportionalControlBasedOnDesignOccupancy",
-              DataSizing::OAFlowCalcMethodNames[static_cast<int>(state->dataMixedAir->VentilationMechanical(1).VentMechZone(1).ZoneOAFlowMethod)]);
+              DataSizing::OAFlowCalcMethodNames[static_cast<int>(ventMechanical.VentMechZone(1).ZoneOAFlowMethod)]);
 
-    state->dataAirLoop->OutsideAirSys.deallocate();
-    state->dataMixedAir->OAMixer.deallocate();
-    state->dataAirLoop->AirLoopZoneInfo.deallocate();
-    state->dataAirSystemsData->PrimaryAirSystems.deallocate();
-    state->dataContaminantBalance->ZoneAirCO2.deallocate();
-    state->dataContaminantBalance->ZoneCO2GainFromPeople.deallocate();
+    state->dataEnvrn->StdRhoAir = 1.2;
+    oaController.MixMassFlow = 1.7 * state->dataEnvrn->StdRhoAir;
+    oaController.MaxOAMassFlowRate = 1.7 * state->dataEnvrn->StdRhoAir;
+    state->dataAirLoop->AirLoopFlow(1).DesSupply = 1.7 * state->dataEnvrn->StdRhoAir;
+
+    // From peeking inside the CO2 limit calcs
+    Real64 zoneCO2Max = 431.08678;
+    Real64 zoneCO2Min = 400.0;
+
+    // Case 1 - Zone CO2 greater than CO2 Max, so OA flow is flow/area+flow/person
+    state->dataContaminantBalance->ZoneAirCO2(1) = 600.0;
+    Real64 expectedOAMassFlow = (oaRequirements.OAFlowPerArea * state->dataHeatBal->Zone(1).FloorArea +
+                                 oaRequirements.OAFlowPerPerson * state->dataHeatBal->Zone(1).TotOccupants) *
+                                state->dataEnvrn->StdRhoAir;
+    oaController.CalcOAController(*state, 1, true);
+    EXPECT_NEAR(expectedOAMassFlow, oaController.OAMassFlow, 0.00001);
+    EXPECT_NEAR(expectedOAMassFlow / oaController.MixMassFlow, oaController.MinOAFracLimit, 0.00001);
+
+    // Case 2 - Zone CO2 greater than CO2 Min, so OA flow is flow/area
+    state->dataContaminantBalance->ZoneAirCO2(1) = 200.0;
+    expectedOAMassFlow = oaRequirements.OAFlowPerArea * state->dataHeatBal->Zone(1).FloorArea * state->dataEnvrn->StdRhoAir;
+    oaController.CalcOAController(*state, 1, true);
+    EXPECT_NEAR(expectedOAMassFlow, oaController.OAMassFlow, 0.00001);
+    EXPECT_NEAR(expectedOAMassFlow / oaController.MixMassFlow, oaController.MinOAFracLimit, 0.00001);
+
+    // Case 3 - Zone CO2 in between CO2 Max and Min, so OA flow is flow/area + proportionate flow/person
+    state->dataContaminantBalance->ZoneAirCO2(1) = zoneCO2Min + 0.3 * (zoneCO2Max - zoneCO2Min);
+    expectedOAMassFlow = (oaRequirements.OAFlowPerArea * state->dataHeatBal->Zone(1).FloorArea +
+                          0.3 * oaRequirements.OAFlowPerPerson * state->dataHeatBal->Zone(1).TotOccupants) *
+                         state->dataEnvrn->StdRhoAir;
+    oaController.CalcOAController(*state, 1, true);
+    EXPECT_NEAR(expectedOAMassFlow, oaController.OAMassFlow, 0.00001);
+    EXPECT_NEAR(expectedOAMassFlow / oaController.MixMassFlow, oaController.MinOAFracLimit, 0.00001);
+}
+
+TEST_F(EnergyPlusFixture, CO2ControlDesignOccupancyTest3Zone)
```

Comment:
3-zone test, identical to single-zone test but with 3 zones, expect 3x OA flow rates.